### ISSUE TITLE
25p movie played at 50hz on TV that's support 25p

### ISSUE THREAD

**Issue by ghost** [koying/SPMC#279]
Sorry for stupid question,
I don't understand why a 25 fps progressive video are played at 50 fps (dubbing the frame)  although my display  devece can handle a 25p and SPMC recognise that.

Video mode detected :

18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 23.976025 Hz
18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 24.000000 Hz
18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with **1920 x 1080 @ 25.000000 Hz**
18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 29.970030 Hz
18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 30.000000 Hz
18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 50.000000 Hz
18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 59.940060 Hz
18:37:28 T:1394252080  NOTICE: Found resolution 1920 x 1080 for display 0 with 1920 x 1080 @ 60.000000 Hz

Video play selected:

18:37:59 T:1480554800  NOTICE:  **fps: 25.000000**, pwidth: 1280, pheight: 720, dwidth: 1280, dheight: 720
18:37:59 T:1480554800  NOTICE: Display resolution ADJUST : **1920x1080 @ 50.000000** - Full Screen (21) (weight: 0.000)

Results: higher CPU/GPU load, 50fps against 25fps on which to apply HQ scaler algorithm....and no tangible improvement for me
Thanks in advance

**Comment by koying** [comment]
> Results: higher CPU/GPU load, 50fps against 25fps on which to apply HQ scaler algorithm....and no tangible improvement for me

Nope. Kodi/SPMC does not "make up" frames, so decoding/rendering CPU usage is the same.
GUI run at 50fps, otoh, which obviously makes it smoother.

You can add the following snippet to advancedsettings.xml to have 25fps at 25Hz

```
  <adjustrefreshrate> 
    <override>
      <fpsmin>24.96</fpsmin>
      <fpsmax>25.01</fpsmax>
      <refreshmin>24.96</refreshmin>
      <refreshmax>25.01</refreshmax>
    </override>
  </adjustrefreshrate>
```

**Comment by ghost** [comment]
Thanks Koying
I added the lines to advancedsettigs.xml files and now the refresh rate is set correctly to 25P when the video frequency is detected to 25,000.

.... but unfortunately there is a new problem ...

I have some video, I suppose, with variable frequency, or so they are detected by codecinfo.
Theystart at 25.000fps, after a minute pass to 24.96fps, then arrived at 25.15fps

this causes a video re sync to 50p, to 25p ,to 50p .... every time the refresh rate exceds the imposed limits.

I set the "adjust the display refresh rate to match video" option to "on the Start / Stop", but it seems not work if there is a ovverride.

I'm wrong? I have to increase fpsmin fpsmax? There is another option to add to the file advancedsettings.xml

Sorry again

**Comment by koying** [comment]
Just change the fpsmin/fpsmax range to, eg, 24.9 to 25.2

**Comment by ghost** [comment]
Thank you very much

**Comment by Bogey** [comment]
I've used the advanced settings, but still the tv displays at 50hz.
Plex does it correct however.
